orttraining/orttraining/core/session/training_session.h

```diff
@@ -283,7 +321,20 @@ class TrainingSession : public InferenceSession {
                                 const std::vector<std::string>& norm_nodes,
                                 const bool dump_convergence_metrics);
 
-  common::Status InsertPipelineOps();
+  // Insert operators for running pipeline and return event tensor names.
+  // For an intermediate pipeline stage, two WaitEvent and two RecordEvent would
+  // be inserted. The dependent event tensor names are returned.
+  // The related computation order is
+  //  WaitEvent --> Forward --> RecordEvent --> WaitEvent --> Backward --> RecordEvent
```

Comment:
It will be changed to `Wait->Recv->Wait->FW->Record->Send->Record->Wait->Recv->Wait->BW->Record->Record`.